wikimedia/wikipedia-ios
# Address and remove notification TODO comments

**Phabricator:** 
https://phabricator.wikimedia.org/T291383

### Notes
This cleans up any TODO comments I had sprinkled throughout the notifications feature. TODOs I addressed are a service extension version check for the push content and adding a couple of missing `didLogIn` notifications to the authentication manager. I didn't feel like the others needed a change, but I'm happy to discuss further.

### Test Steps

Some light regression testing on the service extension is needed to be sure the content is still populating:

1. Enable push notifications via app settings on a device - take the device token sent in the echo subscription API call and use it in our push utility in step 3.
2. Do something on Desktop Wikipedia to trigger a notification.
3. Trigger notification via our push utility.
4. Confirm push content received on device populates with specific notification data, and doesn't just fall back to "New activity on Wikipedia" every time.

Also test that the authentication manager still works:

1. Regression test logging in and out. Be sure notifications bell icon still hides after logout, and shows after login.
2. Test new functionality: Enter airplane mode (and turn off WiFi) before launching the app while logged in. Before the bell icon would not display, even though the app is technically in a logged in state. Now it does show. This is good because there is still useful persisted notifications to display to the user on this screen. Further offline error handling will occur in https://phabricator.wikimedia.org/T287766.
